msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfigurationFactory.kt

```diff
@@ -0,0 +1,266 @@
+        fun initializeNativeAuthConfiguration(context: Context, configResourceId: Int): NativeAuthPublicClientApplicationConfiguration {
+            return initializeNativeAuthConfigurationInternal(context, loadConfiguration(context, configResourceId))
+        }
+
+        /**
+         * Initialize [NativeAuthPublicClientApplicationConfiguration] object from the provided file, if there is any,
+         * and merge it with the default native auth config
+         */
+        @WorkerThread
+        fun initializeNativeAuthConfiguration(context: Context, configFile: File): NativeAuthPublicClientApplicationConfiguration {
+            return initializeNativeAuthConfigurationInternal(context, loadConfiguration(configFile))
+        }
+
+        /**
+         * Initialize the Native Auth configuration with base MSAL default configs and Native Auth default Configs
+         */
+        @WorkerThread
+        private fun initializeNativeAuthConfigurationInternal(context: Context, developerConfig: NativeAuthPublicClientApplicationConfiguration?): NativeAuthPublicClientApplicationConfiguration {
+            // This will create the default msal configuration
+            val defaultMsalConfiguration: PublicClientApplicationConfiguration = PublicClientApplicationConfigurationFactory.loadConfiguration(context, R.raw.msal_default_config)
+
+            // This will create the default native auth configuration
+            val defaultNativeAuthConfiguration = loadDefaultNativeAuthConfiguration(context)
+
+            // Empty configuration to use as basis for merging
+            val config = NativeAuthPublicClientApplicationConfiguration()
+            config.appContext = context
+
+            // Merge base MSAL defaults
+            config.mergeConfiguration(defaultMsalConfiguration)
+            // Marge Native Auth defaults
+            config.mergeConfiguration(defaultNativeAuthConfiguration)
+
+            config.authorities.clear()
+
+            // Check for developerConfig
+            if (developerConfig != null) {
+                // Merge developer Native Auth configuration
+                config.mergeConfiguration(developerConfig)
+                // Validate Native Auth configuration
+                config.validateConfiguration()
+            }
+
+            // Initialize internal library configuration
+            val libraryConfiguration = LibraryConfiguration.builder().authorizationInCurrentTask(config.authorizationInCurrentTask()).build()
+            LibraryConfiguration.intializeLibraryConfiguration(libraryConfiguration)
+
+            config.oAuth2TokenCache = createCache(
+                AndroidPlatformComponentsFactory.createFromContext(context)
+            )
+
+            // Set build values if present in configuration
+            initializeBuildValues(developerConfig)
+
+            return config
+        }
+
+        private fun initializeBuildValues(developerConfig: NativeAuthPublicClientApplicationConfiguration?) {
```

Comment:
I think you should consider removing the BuildValues class entirely. I don't think it is needed given that similar data already exists in BuildConfig, Authority, and AadSliceInfo classes